bundles/org.eclipse.packagedrone.utils.deb/src/org/eclipse/packagedrone/utils/deb/build/DebianPackageWriter.java

```diff
@@ -120,9 +159,15 @@ public void addFile ( final ContentProvider contentProvider, String fileName, En
 
             final TarArchiveEntry entry = new TarArchiveEntry ( fileName );
             entry.setSize ( contentProvider.getSize () );
+            // in case the content provider supplies a modification time itself, use that one
             applyInfo ( entry, entryInformation );
+            // if the modification time should be overridden, then do it here
+            if ( timestampSupplier != null && timestampSupplier.get () != null )
```

Comment:
IMHO this time provider should either be "Optional" or is should be non-null null, checked by `Objects.requireNonNull`.